src/semi_automatic.rs

```diff
@@ -719,14 +704,21 @@ impl<'a> ToTupleImplementation<'a> {
         let trait_ = if let Some(pos) = res
             .attrs
             .iter()
-            .position(|a| a.path.is_ident(TUPLE_TYPES_CUSTOM_TRAIT_BOUND))
+            .position(|a| a.path().is_ident(TUPLE_TYPES_CUSTOM_TRAIT_BOUND))
         {
             // Parse custom trait bound
             let attr = &res.attrs[pos];
-            let input = attr.tokens.to_token_stream();
-            let result = syn::parse2::<BoundsStruct>(input);
+            let Meta::List(items) = &attr.meta else {
+                return Err(Error::new(
+                    attr.span(),
+                    "Expected #[tuple_types_custom_trait_bound($trait_bounds)]",
+                ));
+            };
+
+            let input = items.tokens.to_token_stream();
+            let result = syn::parse2::<syn::TypeTraitObject>(input);
```

Comment:
The tokens no longer include parens by the looks of it, so can parse directly into the target and remove the `BoundsStruct` type.